src/chunkserver/scan_manager.cpp

```diff
@@ -29,148 +30,301 @@ ScanManager::~ScanManager() {
+                    job->type = ScanType::CompareMap;
                     break;
                 }
                 done = true;
                 break;
             case ScanType::CompareMap:
-                CompareMap();
-                if (isCurrentJobFinish()) {
-                    job_.type = ScanType::Finish;
+                CompareMap(key);
+                if (isCurrentJobFinish(key)) {
+                    job->type = ScanType::Finish;
                 } else {
-                    job_.type = ScanType::NewMap;
-                    job_.currentChunkId = GetNextScanChunk();
+                    job->type = ScanType::NewMap;
+                    job->iter++;
+                    job->currentChunkId = job->iter->first;
                 }
                 break;
             case ScanType::Finish:
-                ScanJobFinish();
+                ScanJobFinish(key);
                 done = true;
                 break;
             default:
+                LOG(WARNING) << "Error scan type.";
                 break;
         }
     }
 }
 
-ChunkID ScanManager::GetNextScanChunk() {
-    return 1;
+// send scan request to braft
+void ScanManager::ScanChunkReqProcess(const ScanJob *job) {
+    // split scan chunk request
+    uint32_t currentOffset = 0;
+    auto chunkSize = copysetNodeManager_->GetCopysetNodeOptions().maxChunkSize;
+    auto nodePtr = copysetNodeManager_->GetCopysetNode(job->poolId, job->id);
+    while (currentOffset < chunkSize) {
+        ChunkResponse *response = new ChunkResponse();
+        ChunkRequest request;
+        request.set_optype(CHUNK_OP_TYPE::CHUNK_OP_SCAN);
+        request.set_logicpoolid(job->poolId);
+        request.set_copysetid(job->id);
+        request.set_chunkid(job->currentChunkId);
+        request.set_offset(currentOffset);
+        request.set_size(scanSize_);
+        // reuse the clonefileoffset field to deliver rpc's timeout
+        request.set_clonefileoffset(timeout_);
+
+        std::shared_ptr<ScanChunkRequest> req =
+            std::make_shared<ScanChunkRequest>(nodePtr, this, &request,
+                                               response, nullptr);
+        req->Process();
+        currentOffset += scanSize_;
+    }
 }
 
-void ScanManager::ScanError() {
-    return;
+void ScanManager::SetLocalScanMap(ScanKey key, const ScanMap &map) {
+    ScanJob *job = GetJob(key);
+    if (nullptr != job) {
+        job->localMap = map;
+        job->waitingNum--;
+    }
 }
 
-void ScanManager::ScanChunkReqProcess() {
-    return;
+void ScanManager::DealFollowerScanMap(const FollowScanMapRequest &request,
+                         FollowScanMapResponse *response) {
+    ScanMap scanMap = request.scanmap();
+    ScanKey key(scanMap.logicalpoolid(), scanMap.copysetid());
+    ScanJob *job = GetJob(key);
+    if (nullptr != job) {
+        job->repMap.emplace_back(scanMap);
+        job->waitingNum--;
+        job->type = ScanType::WaitMap;
+        GenScanJob(key);
```

Comment:
> forget set response

here can only set response retcode "CHUNK_OP_STATUS_SUCCESS", because whether the crc is matched should wait two follower send scanmap back.